napari/_qt/qt_main_window.py

```diff
@@ -1594,14 +1634,73 @@ def _screenshot(
                     add_flash_animation(self._qt_viewer._welcome_widget)
             finally:
                 # make sure we always go back to the right canvas size
-                if size is not None or scale is not None:
+                if size is not None or scale is not None or fit_to_data_extent:
                     canvas.size = prev_size
```

Comment:
Why do we need this `if`? Why can we not just assign prev_size?

Reply:
Before we only adjusted this in the conditional of fit_to_data_extent, since that is not the case anymore the conditional is removed.